cli/azd/docs/featue-toggle.md

```diff
@@ -0,0 +1,61 @@
+# Feature Toggle
+
+The Azure Developer CLI includes a `feature toggle` which allows the cli to ship `in-preview` ( _a.k.a. experimental_ ). This document provides the feature design for it.
+
+## Main requirement
+
+As a customer, I can explicitly ask `azd` to unblock and show any _experimental_ functionality. The features becoming available are not finalized, so they can be changed or removed in future releases.
+
+The strategy to ask `azd` to `toggle` between modes is:
+
+> azd config set experimental on
```

Comment:
Can we please think about supporting `AZD_EXPERIMENTAL_<flag>`? This is a common CLI strategy to allow both. Since we will always emit a warning message indicating experimental features being on, I think this resolves the env pollution problem.

The reason to have this is to allow **trial** of new features, before enabling it for a particular machine.